src/target/riscv/riscv.c

```diff
@@ -482,6 +482,21 @@ static void riscv_free_registers(struct target *target)
 	}
 }
 
+static void free_reg_names(struct target *target);
+
+static void free_custom_register_names(struct target *target)
+{
+	RISCV_INFO(info);
+
+	if (!info->custom_register_names.reg_names)
+		return;
+
+	for (unsigned int i = 0; i < info->custom_register_names.num_entries; i++)
+		free((void *)info->custom_register_names.reg_names[i]);
```

Comment:
Curious - is the typecast to `void *` need here? 

If no tool asks for it (compilers, linters, etc.) I recommend to drop it.

```suggestion
		free(info->custom_register_names.reg_names[i]);
```

Reply:
Without the cast any compiler will issue a warning about implicit cast from a const pointer to a non-const one.

Reply:
See my two preceding comments. I believe the pointers should not be declared as const. Therefore the extra `(void *)` cast would not need to be here.